File: iosApp/iosApp/Theme.swift
import SwiftUI
import shared

extension SwiftUI.Color {
    static let colorPrimary = Color(MR.colors().colorPrimary.getUIColor())
    static let onPrimary = Color(MR.colors().colorOnPrimary.getUIColor())
    static let colorSecondary = Color(MR.colors().colorSecondary.getUIColor())
    static let onSecondary = Color(MR.colors().colorOnSecondary.getUIColor())
    static let colorTertiary = Color(MR.colors().colorTertiary.getUIColor())
    static let onTertiary = Color(MR.colors().colorOnTertiary.getUIColor())
    static let surface = Color(MR.colors().colorSurface.getUIColor())
    static let onSurface = Color(MR.colors().colorOnSurface.getUIColor())
    static let background = Color(MR.colors().background.getUIColor())
    static let error = Color(MR.colors().colorError.getUIColor())
    static let onError = Color(MR.colors().colorOnError.getUIColor())
    
    static let gray = Color(MR.colors().gray.getUIColor())
}

File: androidApp/src/main/java/pl/msiwak/multiplatform/android/ui/theme/Theme.kt
package pl.msiwak.multiplatform.android.ui.theme

import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.ui.res.colorResource
import pl.msiwak.multiplatform.commonResources.SR

@Composable
private fun appColorScheme() = lightColorScheme(
    primary = colorResource(id = SR.colors.colorPrimary.resourceId),
    onPrimary = colorResource(id = SR.colors.colorOnPrimary.resourceId),
    secondary = colorResource(id = SR.colors.colorSecondary.resourceId),
    onSecondary = colorResource(id = SR.colors.colorOnSecondary.resourceId),
    tertiary = colorResource(id = SR.colors.colorTertiary.resourceId),
    onTertiary = colorResource(id = SR.colors.colorOnTertiary.resourceId),
    surface = colorResource(id = SR.colors.colorSurface.resourceId),
    onSurface = colorResource(id = SR.colors.colorOnSurface.resourceId),
    background = colorResource(id = SR.colors.background.resourceId),
    error = colorResource(id = SR.colors.colorError.resourceId),
    onError = colorResource(id = SR.colors.colorOnError.resourceId)
)

@Composable
fun AppTheme(
    content: @Composable () -> Unit
) {
    val colorScheme = appColorScheme()

    CompositionLocalProvider(LocalOwnColorScheme provides OwnColorScheme(true)) {
        MaterialTheme(
            colorScheme = colorScheme,
            typography = Typography,
            content = content
        )
    }
}
